bio_pieces/parallel_blast.py

```diff
@@ -0,0 +1,219 @@
+        default=None,
+        help='deprecated'
+    )
+    parser.add_argument(
+        '--logs',
+        help='deprecated'
+    )
+    parser.add_argument(
+        '--outheader',
+        help='Deprecated'
+    )
+    return parser.parse_args()
+
+def parallel_blast(inputfile, outfile, ninst, db, blasttype, task, blastoptions):
+    '''
+    Runs blast commands in parallel on a given fasta file
+
+    :param str inputfile: Input fasta path
+    :param str outfile: Output file path
+    :param int ninst: number of cpus to use if not in PBS or SGE job
+    :param str db: Database path to blast against
+    :param str blasttype: Blast exe to use(blastn, blastx, diamond...)
+    :param str task: Blast task to run with -task option for blasttype or 
+        None if blastx/blastp
+    :param str blastoptions: other options to pass to blast
+    '''
+    blast_path = sh.which(blasttype)
+    args = ['-u', '--pipe', '--block', '10', '--recstart', '>']
+    args += generate_sshlogins(ninst)
+    args += [blast_path]
+    if task is not None:
+        args += ['-task', task]
+    args += ['-db', db, '-max_target_seqs', str(MAX_TARGET_SEQS),
+        '-outfmt', '"'+BLAST_FORMAT+'"'
+    ]
+    args += shlex.split(blastoptions)
+    args += ['-query', '-']
+    #args += ['./t.py']
+    p_sh = sh.Command('parallel')
+    print "[cmd] {0}".format('parallel ' + ' '.join(args))
+    try:
+        p = p_sh(*args, _out=open(outfile,'w'), _in=open(inputfile))
+    except sh.ErrorReturnCode as e:
+        print e.stderr
+        sys.exit(e.exit_code)
+
+def parallel_diamond(inputfile, outfile, ninst, db, task, diamondoptions):
+    '''
+    Runs diamond commands in parallel on a given fasta file
+
+    Will not run more than 1 diamond process per host as diamond utilizes
+    threads better than blast
+
+    :param str inputfile: Input fasta path
+    :param str outfile: Output file path
+    :param int ninst: number of threads to use if not in PBS or SGE job
+    :param str db: Database path to blast against
+    :param str task: blastx or blastp
+    :param str diamondoptions: other options to pass to blast
+    '''
+    '''
+    diamond -task blastx -compress 0 -db /path/nt -o outfile -query inputfile -o outfile
+    my $cmd = "$type $task_option  $options -q  $query -d $db  -o $out"; 
+    '''
+    # This seems kinda stupid that we are just replacing cpu count for each
+    # node with 1, but it is easier than refactoring other code to be better
+    sshlogins = generate_sshlogins(ninst)
+    for i in range(0,len(sshlogins),2):
+        cpu,host = sshlogins[i+1].split('/')
+        sshlogins[i+1] = '1/{0}'.format(host)
+    dmnd_path = sh.which('diamond')
+    args = ['-u', '--pipe', '--block', '10', '--recstart', '>', '--cat']
+    args += sshlogins
+    args += [
+        dmnd_path, task, '--threads', str(ninst), '--db', db, '--query', '{}',
+        '--compress', '0'
+    ] + shlex.split(diamondoptions)
+    d_cmd = sh.Command('parallel')
```

Comment:
I feel like both functions are so similar that they could be merged into a single function as well. I had just forked the original parallel_blast function because it was easier. Not sure if I should just pull this chunk out into a separate function or spend the time to refactor both functions into a single one